pkgs/tools/backup/rsnapshot/default.nix

```diff
@@ -28,7 +13,9 @@ stdenv.mkDerivation rec {
   patchPhase = ''
     substituteInPlace "Makefile.in" --replace \
       "/usr/bin/pod2man" "${perl}/bin/pod2man"
-    patch -p0 <${patch}
+    substituteInPlace "rsnapshot-program.pl" --replace \
+      '$config_file = $default_config_file;' \
+      '$config_file = "/etc/rsnapshot.conf";'
   '';
```

Comment:
It seems to me that `configureFlags = [ "--sysconfdir=/etc" ]` would have the same effect without touching the internals of rsnapshot.

After some (long) investigations, the correct way to build rsnapshot is to use
```nix
{
  # ...
  configureFlags = [ "--sysconfdir=/etc --prefix=/" ];
  makeFlags = [ "DESTDIR=$(out)" ];
}
```
See for example https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/rsnapshot.

For me, the build works with 
```diff
@@ -1,2 +1 @@
-{ fetchurl, stdenv, writeText, perl, openssh, rsync, logger,
-  configFile ? "/etc/rsnapshot.conf" }:
+{ fetchurl, stdenv, writeText, perl, openssh, rsync, logger }:
@@ -4,14 +2,0 @@
-let patch = writeText "rsnapshot-config.patch" ''
---- rsnapshot-program.pl       2013-10-05 20:31:08.715991442 +0200
-+++ rsnapshot-program.pl       2013-10-05 20:31:42.496193633 +0200
-@@ -383,7 +383,7 @@
-       }
-       
-       # set global variable
--      $config_file = $default_config_file;
-+      $config_file = '${configFile}';
- }
- 
- # accepts no args
-'';
-in
@@ -31 +15,0 @@ stdenv.mkDerivation rec {
-    patch -p0 <${patch}
@@ -33,0 +18,4 @@ stdenv.mkDerivation rec {
+  configureFlags = [ "--sysconfdir=/etc --prefix=/" ];
+
+  makeFlags = [ "DESTDIR=$(out)" ];
+
```